test/adapter/json_api/parse_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          class ParseTest < Minitest::Test
+            def setup
+              @hash = {
+                'data' => {
+                  'type' => 'photos',
+                  'id' => 'zorglub',
```

Comment:
Oh, right. Yeah, this PR was initially intended to solve request deserialization only. Note that it would be fairly easy to build upon the `parse` method to parse collections.